algo/src/main/java/org/neo4j/graphalgo/KShortestPathsProc.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2017 "Neo4j, Inc." <http://neo4j.com>
+ *
+ * This file is part of Neo4j Graph Algorithms <http://github.com/neo4j-contrib/neo4j-graph-algorithms>.
+ *
+ * Neo4j Graph Algorithms is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.graphalgo;
+
+import org.neo4j.graphalgo.api.Graph;
+import org.neo4j.graphalgo.core.GraphLoader;
+import org.neo4j.graphalgo.core.ProcedureConfiguration;
+import org.neo4j.graphalgo.core.utils.Pools;
+import org.neo4j.graphalgo.core.utils.ProgressLogger;
+import org.neo4j.graphalgo.core.utils.ProgressTimer;
+import org.neo4j.graphalgo.core.utils.TerminationFlag;
+import org.neo4j.graphalgo.impl.yens.WeightedPathExporter;
+import org.neo4j.graphalgo.impl.yens.YensKShortestPaths;
+import org.neo4j.graphalgo.results.AbstractResultBuilder;
+import org.neo4j.graphdb.Direction;
+import org.neo4j.graphdb.Node;
+import org.neo4j.kernel.api.KernelTransaction;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
+import org.neo4j.logging.Log;
+import org.neo4j.procedure.*;
+
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ * Yen's K shortest paths algorithm. Computes multiple shortest
+ * paths from a given start to a goal node in the desired direction.
+ * The paths are written to the graph using new relationships named
+ * by prefix + index.
+ *
+ * @author mknblch
+ */
+public class KShortestPathsProc {
+
+    public static final String DEFAULT_TARGET_PROPERTY = "PATH_";
+    public static final String PREFIX_IDENTIFIER = "writePropertyPrefix";
+
+    @Context
+    public GraphDatabaseAPI api;
+
+    @Context
+    public Log log;
+
+    @Context
+    public KernelTransaction transaction;
+
+    @Procedure(value = "algo.kShortestPaths", mode = Mode.WRITE)
+    @Description("CALL algo.shortestPath(startNode:Node, endNode:Node, k:int, weightProperty:String" +
```

Comment:
I think the description should say `CALL algo.kShortestPaths`?

Reply:
yep thx